Cannot add Teams Receiver with AlertmanagerConfig

**What happened?**
I'm trying to add a Teams Receiver using the AlertmanagerConfig CRD.
I get the following error (Operator logs). I use the kube-prometheus stack.
Hope I'm right here anyway, because the error comes directly from the Prometheus operator? 


**Environment**

* Prometheus Operator version:

    `v0.69.1`


* Manifests:

```
apiVersion: monitoring.coreos.com/v1alpha1
kind: AlertmanagerConfig
metadata:
  name: alertmanager-config
  labels:
    resource: prometheus
spec:
  route:
    groupBy: ["..."]
    groupWait: 1m
    groupInterval: 1m
    repeatInterval: 12h
    receiver: "default_channel"
    routes:
      - matchers:
        - matchType: =
          name: severity
          value: warning
        receiver: teams
receivers:
    - name: "default_channel"
    - name: teams
      msteamsConfigs:
        - webhookUrl:
            name: alertmanager-teams
            key: webhook
```

* Prometheus Operator Logs:

```
panic: runtime error: invalid memory address or nil pointer dereference
 [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1a3ead3]                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                  
 goroutine 345 [running]:                                                                                                                                                                                                                                                                                         
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).convertMSTeamsConfig(0xc000cac3b8, {0x25a6830, 0xc000675090}, {0x0, {{{0xc0008748b8, 0x12}}, {0xc001424a50, 0x7}, 0x0}, 0x0, ...}, ...)                                                                                     
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:1259 +0x93                                                                                                                                                                                                               
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).convertReceiver(0xc0003e2df0?, {0x25a6830, 0xc000675090}, 0xc000cac010, {{0xc001424a06?, 0xc000cabf30?}, {0xc000874888?, 0xc0006fa0f0?}})                                                                                   
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:646 +0x229f                                                                                                                                                                                                              
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).initializeFromAlertmanagerConfig(0xc000cac3b8, {0x25a6830, 0xc000675090}, 0x0?, 0xc00078e840)                                                                                                                               
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:241 +0x372                                                                                                                                                                                                               
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).provisionAlertmanagerConfiguration(0xc00094e300, {0x25a6830, 0xc000675090}, 0xc000a3a000, 0xc000f8eb40)                                                                                                                          
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:951 +0x5c5                                                                                                                                                                                                            
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).sync(0xc00094e300, {0x25a6830, 0xc000675090}, {0xc000936440, 0x32})                                                                                                                                                              
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:663 +0x6a8                                                                                                                                                                                                            
 github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).Sync(0xc00094e300, {0x25a6830?, 0xc000675090?}, {0xc000936440, 0x32})                                                                                                                                                            
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:618 +0x2c                                                                                                                                                                                                             
 github.com/prometheus-operator/prometheus-operator/pkg/operator.(*ResourceReconciler).processNextReconcileItem(0xc0008ea000, {0x25a6830, 0xc000675090})                                                                                                                                                          
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/operator/resource_reconciler.go:396 +0x22e                                                                                                                                                                                                     
 github.com/prometheus-operator/prometheus-operator/pkg/operator.(*ResourceReconciler).Run.func1()                                                                                                                                                                                                                
     /home/runner/work/prometheus-operator/prometheus-operator/pkg/operator/resource_reconciler.go:359 +0x3f                                                                                                                                                                                                      
 golang.org/x/sync/errgroup.(*Group).Go.func1()                                                                                                                                                                                                                                                                   
     /home/runner/go/pkg/mod/golang.org/x/sync@v0.4.0/errgroup/errgroup.go:75 +0x56                                                                                                                                                                                                                               
 created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 52                                                                                                                                                                                                                                                
     /home/runner/go/pkg/mod/golang.org/x/sync@v0.4.0/errgroup/errgroup.go:72 +0x96
```

--- a/pkg/alertmanager/amcfg.go
+++ b/pkg/alertmanager/amcfg.go
@@ -1256,8 +1256,14 @@ func (cb *configBuilder) convertMSTeamsConfig(
 ) (*msTeamsConfig, error) {
 	out := &msTeamsConfig{
 		SendResolved: in.SendResolved,
-		Title:        *in.Title,
-		Text:         *in.Text,
+	}
+
+	if in.Title != nil {
+		out.Title = *in.Title
+	}
+
+	if in.Text != nil {
+		out.Text = *in.Text
 	}
 
 	webHookURL, err := cb.store.GetSecretKey(ctx, crKey.Namespace, in.WebhookURL)

--- a/pkg/alertmanager/amcfg_test.go
+++ b/pkg/alertmanager/amcfg_test.go
@@ -1965,6 +1965,56 @@ func TestGenerateConfig(t *testing.T) {
 			},
 			golden: "CR_with_MSTeams_Receiver.golden",
 		},
+		{
+			name:      "CR with MSTeams Receiver with Partial Conf",
+			amVersion: &version26,
+			kclient: fake.NewSimpleClientset(
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "ms-teams-secret",
+						Namespace: "mynamespace",
+					},
+					Data: map[string][]byte{
+						"url": []byte("https://webhook.office.com/webhookb2/id/IncomingWebhook/id"),
+					},
+				},
+			),
+			baseConfig: alertmanagerConfig{
+				Route: &route{
+					Receiver: "null",
+				},
+				Receivers: []*receiver{{Name: "null"}},
+			},
+			amConfigs: map[string]*monitoringv1alpha1.AlertmanagerConfig{
+				"mynamespace": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "myamc",
+						Namespace: "mynamespace",
+					},
+					Spec: monitoringv1alpha1.AlertmanagerConfigSpec{
+						Route: &monitoringv1alpha1.Route{
+							Receiver: "test",
+						},
+						Receivers: []monitoringv1alpha1.Receiver{
+							{
+								Name: "test",
+								MSTeamsConfigs: []monitoringv1alpha1.MSTeamsConfig{
+									{
+										WebhookURL: corev1.SecretKeySelector{
+											Key: "url",
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "ms-teams-secret",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			golden: "CR_with_MSTeams_Receiver_Partial_Conf.golden",
+		},
 	}
 
 	logger := log.NewNopLogger()

--- a/pkg/alertmanager/testdata/CR_with_MSTeams_Receiver_Partial_Conf.golden
+++ b/pkg/alertmanager/testdata/CR_with_MSTeams_Receiver_Partial_Conf.golden
@@ -1,0 +1,13 @@
+route:
+  receiver: "null"
+  routes:
+  - receiver: mynamespace/myamc/test
+    matchers:
+    - namespace="mynamespace"
+    continue: true
+receivers:
+- name: "null"
+- name: mynamespace/myamc/test
+  msteams_configs:
+  - webhook_url: https://webhook.office.com/webhookb2/id/IncomingWebhook/id
+templates: []

